### ISSUE TITLE
AutoRest crashes for `application/xml` encoded responses in combination with `x-ms-parameter-grouping`.

### ISSUE THREAD

**Issue by dlgit** [Azure/autorest#2236]
**Background**:

I have an operation in webapi service like follows:
  ```
[Route("books")]
        [HttpGet]
        [ResponseType(typeof(GetBooksResponse))]
        public IHttpActionResult GetBooks([FromUri] GetBooksRequest request)
        {
                ...
        }
```

```
GetBooksRequest 
{
  Genre,
  Publisher,
  ...
}
```

Swagger spec:

```
 "parameters": [
         {
            "name": "Genre",
            "in": "query",
            "required": false,
            "type": "string"
          },
          {
            "name": "Publisher",
            "in": "query",
            "required": false,
            "type": "string"
          },
...
```

I would like to use **x-ms-parameter-grouping** to group method parameters Genre,Publisher, etc in auto generated client, and wrote a ParameterGroupOperationFilter

```
public class ParameterGroupOperationFilter : IOperationFilter
    {
        public void Apply(Operation operation, SchemaRegistry schemaRegistry, ApiDescription apiDescription)
        {
            if (operation.parameters == null)
                return;

            var parameters = apiDescription.ActionDescriptor.GetParameters();
            if (parameters?.Count == 1))
            {
                foreach (var parm in operation.parameters)
                {
                    if ((parm.@in).Equals("path") || (parm.@in).Equals("query"))
                    {
                        parm.vendorExtensions.Add("x-ms-parameter-grouping",
                                            new { name = parameters[0].ParameterType.Name + "parameter" });
                    }

                }
                   
            }
        }
    }
```

When I use this operationfilter in Swagger Config, and apply AutoRest 1.0.1.0, it throws an error:
 
![image](https://cloud.githubusercontent.com/assets/10066541/25668371/801d8764-2ff4-11e7-8ea8-6557e5ba0547.png)

BTW, when I use AutoRest 0.17.0.0, it works fine.

**Question**: Is there anything missing in ParameterGroupOperationFilter? What is that error about?

**Comment by olydis** [comment]
What is a `ParameterGroupOperationFilter` or `IOperationFilter`? How do you use it in which "Swagger Config"? Can you show us the acutal Swagger that caused the error you posted?

**Comment by dlgit** [comment]
After I apply ParameterGroupOperationFilter, the Swagger is like this:
 {
            "name": "Gnere",
            "in": "query",
            "required": false,
            "type": "string",
            "x-ms-parameter-grouping": {
              "name": "GetBooksRequestparameter"
            }
          },
          {
            "name": "Publisher",
            "in": "query",
            "required": false,
            "type": "string",
            "x-ms-parameter-grouping": {
              "name": "GetBooksRequestparameter"
            }
          },

**Comment by dlgit** [comment]
Swagger:

 "paths": {
   
    "/books": {
      "get": {
        "tags": [
          "Books"
        ],
        "summary": "Gets books",
        "operationId": "Members_GetBooks",
        "consumes": [],
        "produces": [
          "application/json",
          "text/json",
          "application/xml",
          "text/xml"
        ],
        "parameters": [
          {
            "name": "Genre",
            "in": "query",
            "required": false,
            "type": "string",
            "x-ms-parameter-grouping": {
              "name": "GetBooksRequestparameter"
            }
          },
          {
            "name": "Publisher",
            "in": "query",
            "required": false,
            "type": "string",
            "x-ms-parameter-grouping": {
              "name": "GetBooksRequestparameter"
            }
          },
          {
            "name": "Accept",
            "in": "header",
            "description": "The version of the call to the API to make.",
            "required": false,
            "schema": {
              "default": "application/json; version=1",
              "type": "string",
              "readOnly": true
            },
            "type": "string",
            "default": "application/json; version=1"
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/GetProfilesResponse"
            }
          }
        }
      }
    }
  },
  "definitions": {
   
    "GetBooksRequest": {
      "type": "object",
      "allOf": [
        {
          "$ref": "#/definitions/ServiceRequest"
        }
      ],
      "properties": {
        
        "Genre": {
          "type": "string",
          "x-ms-parameter-grouping": {
            "name": "GetBooksRequestparameter"
          }
        },
        "Publisher": {
          "type": "string",
          "x-ms-parameter-grouping": {
            "name": "GetBooksRequestparameter"
          },
        "ApplicationName": {
          "type": "string",
          "x-ms-parameter-grouping": {
            "name": "GetBooksRequestparameter"
          }
        }
      }
    },
    "GetBooksResponse": {
      "type": "object",
      "properties": {
        "Books": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Book"
          }
        }
       ...
        }
      }
    },
    "ServiceRequest": {
      "required": [
        "$type"
      ],
      "type": "object",
      "properties": {
        "ApplicationName": {
          "type": "string"
        },
        "$type": {
          "type": "string"
        }
      },
      "discriminator": "$type"
    }
   
}

**Comment by olydis** [comment]
Thanks for the Swagger. The problem has nothing to do with `ParameterGroupOperationFilter` or similar, we simply seem to have a bug in our XML serialization related code.

Workaround: Remove the `application/xml` and `text/xml` from the `produces` section. AutoRest then generates code for me. 🙂 

Of course, this means that the generated client won't accept XML responses by the server - we'll fix that bug!

**Comment by olydis** [comment]
fixed with #2237, should make it into the nightly builds soon.

**Comment by dlgit** [comment]
Thank you very much!
